### PR TITLE
The GameLauncher now starts at the screen resolution

### PR DESCRIPTION
**Pull request by galibzon**
## What does this PR do?
The GameLauncher now starts at the screen resolution
and fullscreen mode as specified in the command line via the CVARs: r_width, r_height, r_fullscreen.

REMARK: It was found that Atom's BootstrapSystemComponent::Activate() is called BEFORE Legacy/System.cpp::ExecuteCommandLine(), the solution for now is to our own command line parsing for the required CVARs.

## How was this PR tested?
Validated that AutomatedTesting.GameLauncher starts at the request screen resolution specified by --r_width and --r_height when given in the command line arguments.
Also made sure it enters fullscreen mode if --r_fullscreen=true

Signed-off-by: galibzon <66021303+galibzon@users.noreply.github.com>

### CODE REVIEW

File: Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp
Comment (rgba16f) [inline Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp:140]: Not a blocker but it would be really nice if these command line had a functions like this to avoid all this boiler plate code to query the value.
```
     bool GetBoolValue(AZStd::string_view switchName, bool default value = false);
     uint32_t GetUInt32Value(AZStd::string_view switchName, uint32_t default value = 1);
```

**Review by rgba16f — APPROVED** [review]

File: Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp
Comment (galibzon) [reply Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp:140]: Was thinking about that too. Will leave it for later.

**Comment by galibzon** [comment]
> Does this change also work with values passed in a cfg file or a setreg value?

It's only for changes coming from the command line. CFG parsing seems to occur also afterwards. In other words, the clean solution is to fix the order of instantiation so System.cpp gets to read command line arguments + CFG file data, then later the BootstrapSystemcomponent only has to read the CVAR values.

File: Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp
Comment (moraaar) [inline Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp:51]: It seems more appropriate to use setting registry for these so they can be put in .setreg files. And more specifically platform dependent setting registry (read with `AZ::SettingsRegistryMergeUtils::PlatformGet`), so different values per platform could be specified.

These CVars will appear on the console and could be modified at runtime, but they won't make any difference.

File: Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp
Comment (moraaar) [inline Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp:199]: how will affect calling `SetFullScreenState` on mobile platforms? Is this suppose to be called on host platforms only?

**Comment by moraaar** [comment]
Is there some way to do resolution scaling at the moment? If yes, how does code change works with it?

File: Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp
Comment (lumberyard-employee-dm) [reply Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp:51]: Actually `PlatformGet` should not be used.
That was added just for porting over the bootstrap.cfg settings.

The recommended way to specify platform dependent settings registry settings is by adding a `Platform/<platform-name>` folder underneath the `Registry` directory that is being used (https://www.o3de.org/docs/user-guide/settings/developer-documentation/#merge-settings-files)

Comment (moraaar) [reply Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp:51]: Thanks for the clarification @lumberyard-employee-dm. I wasn't away of that recommendation.

**Comment by galibzon** [comment]
> Is there some way to do resolution scaling at the moment? If yes, how does this code work in conjunction with it?

This is a quick workaround (aka Manager's code) to unblock performance collection on Windows & Linux. I opened a ticket to address fullscreen issues: https://github.com/o3de/o3de/issues/13147

**Review by antonmic — APPROVED** [review]

File: Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp
Comment (galibzon) [reply Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp:51]: I understand your concern, and the problem we have is that there's no unified way at the moment. There are things in CVars, others in Registry settings, others in CFG files, etc. I'll leave it to the engineers to eventually clean that up. For now, this PR enables screen resolution setup while minimizing verbosity.

File: Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp
Comment (galibzon) [reply Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp:199]: See my comment below.

File: Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp
Comment (galibzon) [reply Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp:195]: Let us know when it gets fixed and we'll make changes to this code. Yes, I tried that API and it didn't work.

**Pull request merged**
